spring-cloud-bindings/src/main/java/org/springframework/cloud/bindings/boot/SpringBootVersionResolver.java

```diff
@@ -0,0 +1,23 @@
+package org.springframework.cloud.bindings.boot;
+
+import org.springframework.boot.SpringBootVersion;
+
+public class SpringBootVersionResolver {
+
+    private int forcedVersion;
+
+    public SpringBootVersionResolver() {
+    }
+
+    protected SpringBootVersionResolver(int forcedVersion) {
+        this.forcedVersion = forcedVersion;
+    }
+
+    public boolean isBootMajorVersionEnabled(int bootVersion) {
+        if (forcedVersion != 0) {
+            return forcedVersion == bootVersion;
+        }
+        int major = Integer.parseInt(SpringBootVersion.getVersion().split("\\.")[0]);
+        return major == bootVersion;
+    }
+}
```

Comment:
**[issue]** Make this package-protected ; you don't want this class to be a part of your public API (same as `MapMapper`).
**[suggestion]** set the default value for `forcedVersion` to `-1`, it's a usual convention for "this is invalid".
**[suggestion]** Maybe add a comment, "this class is used in unit tests, to fake the resolution of which major version of spring boot is used" or something similar.

```suggestion
class SpringBootVersionResolver {

    private int forcedVersion = -1;

    SpringBootVersionResolver() {
    }

    protected SpringBootVersionResolver(int forcedVersion) {
        this.forcedVersion = forcedVersion;
    }

   protected boolean isBootMajorVersionEnabled(int bootVersion) {
        if (forcedVersion != -1) {
            return forcedVersion == bootVersion;
        }
        int major = Integer.parseInt(SpringBootVersion.getVersion().split("\\.")[0]);
        return major == bootVersion;
    }
}
```